### PR TITLE
Implements _pretty parameter

### PR DESCRIPTION
**Pull request by rotodd**
## Description
- Enables clients to request pretty-printed resources using the _pretty parameter as described in the [spec](https://www.hl7.org/fhir/http.html#parameters)
- Adds GetIsPretty extension method which pulls the _pretty parameter value from the request query string
- Adds indentation to the response if GetIsPretty returns true

## Related issues
Addresses [AB#69450](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/69450).

## Testing
- Adds new Unit tests to ensure GetIsPretty extension method returns expected values

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs [hunk 1]
```diff
@@ -37,5 +38,33 @@ public static SummaryType GetSummaryType(this HttpContext context, ILogger logge
...
+                    {
+                        // Assume no pretty formatting if parameter can't be parsed.
+                        isPretty = default;
+                    }
+
+                    logger.LogDebug("Changing response pretty formatting to '{0}'", isPretty);
+
+                    return isPretty;
+                }
+                catch (Exception ex)
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs]: I think we can forgo the exception logging here. It should be getting logged even without this.

File: src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_PRETTY", "TRUE");
+
+            var isPretty = context.GetIsPretty(_logger);
+
+            Assert.True(isPretty);
+        }
+
+        [Fact]
+        public void GivenARequestWithPrettyIndentationSetToFalseInCaps_WhenSerializingTheResponse_ThenPrettyIndentationIsNotApplied()
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs]: These are good tests. You could use xUnit's "theories" to parameterize the same test with different inputs and expected outputs to save some typing and space. Take a look at DecimalExtensionTests.cs

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs (hunk 1 above)
Comment (rotodd) [reply src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs]: Good to know. Removed in d40c88d.

File: src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs (hunk 2 above)
Comment (rotodd) [reply src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs]: Oh cool, thanks. Let me know what you think of 395dd9b.

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs [hunk 3]
```diff
@@ -37,5 +38,25 @@ public static SummaryType GetSummaryType(this HttpContext context, ILogger logge
...
+
+            if (!string.IsNullOrWhiteSpace(query))
+            {
+                if (!bool.TryParse(query, out bool isPretty))
+                {
+                    // Assume no pretty formatting if parameter can't be parsed.
+                    isPretty = default;
+                }
+
+                logger.LogDebug("Changing response pretty formatting to '{0}'", isPretty);
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs]: I would not log here. This get method would ideally be free of side-effects (and you are not "changing" anything here).

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs [hunk 4]
```diff
@@ -8,10 +8,11 @@
 using Hl7.Fhir.Rest;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Features;
 
 namespace Microsoft.Health.Fhir.Api.Features.Formatters
 {
-    public static class SummaryTypeExtensions
+    public static class HttpExtensions
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs]: Maybe HttpContextExtensions?

File: src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs (hunk 2 above)
Comment (johnstairs) [reply src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs]: Looks good!

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs (hunk 4 above)
Comment (rotodd) [reply src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs]: Sounds good, fixed in ae3c2db and 9cdb676.

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs (hunk 3 above)
Comment (rotodd) [reply src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs]: Makes sense, thanks. Removed in 827c3e5.

**Pull request merged**